File: src/main/java/org/algohub/engine/judge/CreateFriendlyMessage.java
package org.algohub.engine.judge;

class CreateFriendlyMessage {

    String from(final String errorMessage) {
        final StringBuilder sb = new StringBuilder();
        final String[] lines = errorMessage.split("\n");
        for (final String line : lines) {
            final int pos = line.indexOf(".java:");
            if (pos > 0) {
                appendFriendlyMessage(sb, line, pos);
            } else {
                sb.append(line).append('\n');
            }
        }
        return sb.toString();
    }

    private void appendFriendlyMessage(StringBuilder sb, String line, int pos) {
        final int pos2 = line.indexOf(':', pos + ".java:".length());
        final String friendlyMessage = String.format(
                "Line:%d%s",
                getLineNumber(line, pos, pos2),
                line.substring(pos2)
        );
        sb.append(friendlyMessage).append('\n');
    }

    private int getLineNumber(String line, int pos, int pos2) {
        final int lineNumber;
        {
            final String numberStr = line.substring(pos + ".java:".length(), pos2);
            lineNumber = Integer.valueOf(numberStr);
        }
        return lineNumber;
    }
}

File: src/main/kotlin/org/algohub/engine/judge/CreateFriendlyMessage.kt
package org.algohub.engine.judge

internal class CreateFriendlyMessage {

    private val JAVA_EXT = ".java:"

    fun from(errorMessage: String): String {
        val lines = errorMessage
                .split("\n")
                .dropLastWhile(String::isEmpty)
                .toTypedArray()

        val sb = StringBuilder()
        for (line in lines) {
            processLine(sb, line)
        }

        return sb.toString()
    }

    private fun processLine(sb: StringBuilder, line: String) {
        val pos = line.indexOf(JAVA_EXT)

        if (pos > 0) {
            appendFriendlyMessage(sb, line, pos)
        } else {
            sb.append(line).append('\n')
        }
    }

    private fun appendFriendlyMessage(sb: StringBuilder, line: String, pos: Int) {
        val pos2 = line.indexOf(':', pos + JAVA_EXT.length)
        val friendlyMessage = "Line:${getLineNumber(line, pos, pos2)}${line.substring(pos2)}"
        sb.append(friendlyMessage).append('\n')
    }

    private fun getLineNumber(line: String, pos: Int, pos2: Int): Int {
        val numberStr = line.substring(pos + JAVA_EXT.length, pos2)
        return Integer.valueOf(numberStr)!!
    }
}
